src/react-apps/applications/runtime/src/features/instantiate/containers/PartySelection.tsx

```diff
@@ -204,11 +181,27 @@ function PartySelection(props: IPartySelectionProps) {
     if (!language.party_selection) {
       return null;
     }
-    return `
-      ${language.party_selection.invalid_selection_first_part} ${getRepresentedPartyName()}.
-      ${language.party_selection.invalid_selection_second_part} ${templatePartyTypesString()}.
-      ${language.party_selection.invalid_selection_third_part}
-    `;
+    if (location.state !== undefined &&
+      (location.state as IRedirectReason) &&
+      (location.state as IRedirectReason) !== undefined) {
+      switch ((location.state as IRedirectReason).errorType) {
```

Comment:
SonarCloud code smell... why the switch statement here?

Reply:
I feel that the switch statement is easier to read, and it makes it easier to expand when we need more "error"-handling for what I call "Redirect to partyselection reasons".

And it's only readability code smell. Nothing major.

Reply:
Is there coming more errors?

Reply:
Most likely. I think we need a message if the user has checked "always show me partyselection", even though they are representing a valid party.

And it could be nice to have a message that says "You see this page because you have checked "Always show party selection" in preferences".